browser/ui/browser_commands.cc

```diff
@@ -81,17 +82,36 @@ void OpenGuestProfile() {
   profiles::SwitchToGuestProfile(ProfileManager::CreateCallback());
 }
 
-void ToggleSpeedreader(Browser* browser) {
+void MaybeDistillAndShowSpeedreaderBubble(Browser* browser) {
 #if BUILDFLAG(ENABLE_SPEEDREADER)
   speedreader::SpeedreaderService* service =
       speedreader::SpeedreaderServiceFactory::GetForProfile(browser->profile());
   if (service) {
-    // This will trigger a button update via a pref change subscribition.
-    service->ToggleSpeedreader();
-
     WebContents* contents = browser->tab_strip_model()->GetActiveWebContents();
     if (contents) {
-      contents->GetController().Reload(content::ReloadType::NORMAL, false);
+      auto* tab_helper =
+          speedreader::SpeedreaderTabHelper::FromWebContents(contents);
+      if (!tab_helper)
+        return;
+
+      const bool speedreader_enabled = tab_helper->IsSpeedreaderEnabled();
+      const DistillState state = tab_helper->PageDistillState();
+
+      if (state == DistillState::kSpeedreaderMode) {
+        tab_helper->ShowSpeedreaderBubble();
+      } else if (state == DistillState::kReaderMode) {
+        tab_helper->ShowReaderModeBubble();
```

Comment:
If I disable speedreader for a single site (when speedreader is enabled globally), I can get there. It shows "Turn on speedreader?", I click, and it disables the global pref

Reply:
fixed the state logic and commented it.